tiatoolbox/annotation/storage.py

```diff
@@ -0,0 +1,1709 @@
+                The predicate to evaulate against candidate properties
+                during the query.
+
+        Returns:
+            sqlite3.Cursor: A database cursor for the current query.
+
+        """
+        query_geometry = geometry
+        if select_callable is None:
+            select_callable = select
+        if geometry_predicate not in self._geometry_predicate_names:
+            raise ValueError(
+                "Invalid geometry predicate."
+                f"Allowed values are: {', '.join(self._geometry_predicate_names)}."
+            )
+        cur = self.con.cursor()
+        if isinstance(query_geometry, Iterable):
+            query_geometry = Polygon.from_bounds(*query_geometry)
+        min_x, min_y, max_x, max_y = query_geometry.bounds
+
+        if isinstance(where, Callable):
+            select = select_callable
+
+        query_string = (
+            "SELECT "  # skipcq: BAN-B608
+            + select  # skipcq: BAN-B608
+            + """
+         FROM annotations, rtree
+        WHERE annotations.id == rtree.id
+          AND max_x >= :min_x
+          AND min_x <= :max_x
+          AND max_y >= :min_y
+          AND min_y <= :max_y
+          AND geometry_predicate(:geometry_predicate, :query_geometry, boundary, cx, cy)
+        """
+        )
+        query_parameters = {
+            "min_x": min_x,
+            "max_x": max_x,
+            "min_y": min_y,
+            "max_y": max_y,
+            "geometry_predicate": geometry_predicate,
+            "query_geometry": query_geometry.wkb,
+        }
+        if isinstance(where, str):
+            sql_predicate = eval(where, SQL_GLOBALS, {})  # skipcq: PYL-W0123
+            query_string += f"AND {sql_predicate}"
+        if isinstance(where, bytes):
+            query_string += "AND pickle_where(:where, properties)"
+            query_parameters["where"] = where
+        cur.execute(query_string, query_parameters)
+        return cur
+
+    def iquery(
+        self,
+        geometry: QueryGeometry,
+        where: Union[str, bytes, Callable[[Geometry, Dict[str, Any]], bool]] = None,
+        geometry_predicate="intersects",
+    ) -> List[str]:
+        query_geometry = geometry
+        cur = self._query(
+            "[key]",
+            geometry=query_geometry,
+            geometry_predicate=geometry_predicate,
+            where=where,
+            select_callable="[key], boundary, properties",
+        )
+        if isinstance(where, Callable):
+            return [
+                key
+                for key, _, properties in cur.fetchall()
+                if where(json.loads(properties))
+            ]
+        return [key for key, in cur.fetchall()]
+
+    def query(
+        self,
+        geometry: QueryGeometry,
+        where: Union[str, bytes, Callable[[Geometry, Dict[str, Any]], bool]] = None,
+        geometry_predicate: str = "intersects",
+    ) -> List[Annotation]:
+        query_geometry = geometry
```

Comment:
docstring missing.

Reply:
Docstring should be inhereted, see other comment.

Reply:
ok no problem. It's fine if it's inheriting it.